### ISSUE TITLE
Most appropriate way to change device name?

### ISSUE THREAD

**Issue by infovore** [adafruit/Adafruit_CircuitPython_MIDI#31]
I've been prodding this library as an alternative to some other code I have, and it's super-impressive. One question: what's the most appropriate way to change the auto-generated device name?

In my MIDI Monitor tool, I get this:

![image](https://user-images.githubusercontent.com/1103/111634708-ad09d100-87ee-11eb-935c-2e49684f0e92.png)

and I've had good success with other boards on changing the detected devicename - which I appreciate might be a USB setting rather than a MIDI-specific setting. I'm new to CP, so am imagining this is a `settings.py` or `boot.py` issue, but don't quite know where to begin.

**Comment by dhalbert** [comment]
This is set in CIrcuitPython at compile time: https://github.com/adafruit/circuitpython/blob/11e510a06ab67b5b78349c91d38ea780f255a425/tools/gen_usb_descriptor.py#L398-L458, so you would need to do a custom build of CircuitPython to change it.

We are going to work on dynamic USB descriptors, so setting this name could be a capability we would add. It would not be in this library. If you are interested in this, I will migrate this issue to the `circuitpython` repo.

**Comment by infovore** [comment]
@dhalbert yes, that sounds like a good fit - I can see now it's part of the `usbmidi` code. It'd be a nice-to-have alongside custom USB descriptors (which I've used a lot on other prototyping boards as part of products).
